Flaky test - Test_PushMetrics/WAL/unmatchedBoundBucketHist_case

Seen here: https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/5460547254?check_suite_focus=true

```
--- FAIL: Test_PushMetrics (69.81s)
    --- FAIL: Test_PushMetrics/WAL (0.00s)
        --- PASS: Test_PushMetrics/WAL/staleNaNSummary_case (4.11s)
        --- PASS: Test_PushMetrics/WAL/doubleSum_case (8.21s)
        --- PASS: Test_PushMetrics/WAL/staleNaNEmptyHistogram_case (8.20s)
        --- PASS: Test_PushMetrics/WAL/staleNaNSum_case (8.21s)
        --- PASS: Test_PushMetrics/WAL/staleNaNIntSum_case (8.21s)
        --- PASS: Test_PushMetrics/WAL/staleNaNHistogram_case (8.20s)
        --- PASS: Test_PushMetrics/WAL/staleNaNDoubleGauge_case (8.20s)
        --- PASS: Test_PushMetrics/WAL/emptySummary_case (8.20s)
        --- PASS: Test_PushMetrics/WAL/staleNaNIntGauge_case (8.20s)
        --- PASS: Test_PushMetrics/WAL/emptyCumulativeSum_case (8.21s)
        --- PASS: Test_PushMetrics/WAL/emptyGauge_case (8.21s)
        --- PASS: Test_PushMetrics/WAL/5xx_case (8.21s)
        --- FAIL: Test_PushMetrics/WAL/unmatchedBoundBucketHist_case (8.22s)
```

cc @Aneurysm9

--- a/exporter/prometheusremotewriteexporter/exporter_test.go
+++ b/exporter/prometheusremotewriteexporter/exporter_test.go
@@ -656,11 +656,11 @@ func Test_PushMetrics(t *testing.T) {
 					prwe, nErr := newPRWExporter(cfg, set)
 					require.NoError(t, nErr)
 					ctx, cancel := context.WithCancel(context.Background())
-					t.Cleanup(cancel)
+					defer cancel()
 					require.NoError(t, prwe.Start(ctx, componenttest.NewNopHost()))
-					t.Cleanup(func() {
+					defer func() {
 						require.NoError(t, prwe.Shutdown(ctx))
-					})
+					}()
 					err := prwe.PushMetrics(ctx, *tt.md)
 					if tt.returnErr {
 						assert.Error(t, err)

--- a/exporter/prometheusremotewriteexporter/wal.go
+++ b/exporter/prometheusremotewriteexporter/wal.go
@@ -159,18 +159,21 @@ func (prwe *prweWAL) run(ctx context.Context) (err error) {
 		return
 	}
 
+	runCtx, cancel := context.WithCancel(ctx)
+
 	// Start the process of exporting but wait until the exporting has started.
 	waitUntilStartedCh := make(chan bool)
 	go func() {
 		signalStart := func() { close(waitUntilStartedCh) }
 		for {
 			select {
-			case <-ctx.Done():
+			case <-runCtx.Done():
 				return
 			case <-prwe.stopChan:
+				cancel()
 				return
 			default:
-				err := prwe.continuallyPopWALThenExport(ctx, signalStart)
+				err := prwe.continuallyPopWALThenExport(runCtx, signalStart)
 				signalStart = func() {}
 				if err != nil {
 					// log err

